From: Beniamino Galvani <bgalvani@redhat.com>
Date: Fri, 10 Jun 2016 14:03:10 +0200
Subject: secret-agent: increase timeout for GetSecrets D-Bus call

The default value of 25 seconds is too short, bump it to 120.


Bug: https://bugzilla.gnome.org/show_bug.cgi?id=767321
Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1349740
Bug-Ubuntu: https://bugs.launchpad.net/canonical-devices-system-image/+bug/1588126
---
 src/settings/nm-secret-agent.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/settings/nm-secret-agent.c b/src/settings/nm-secret-agent.c
index bfa7b50..8bc88fd 100644
--- a/src/settings/nm-secret-agent.c
+++ b/src/settings/nm-secret-agent.c
@@ -362,6 +362,9 @@ nm_secret_agent_get_secrets (NMSecretAgent *self,
 	r = request_new (self, "GetSecrets", path, setting_name, callback, callback_data);
 	r->is_get_secrets = TRUE;
 	g_hash_table_add (priv->requests, r);
+
+	/* Increase the timeout only for this call */
+	g_dbus_proxy_set_default_timeout (G_DBUS_PROXY (priv->proxy), 120000);
 	nmdbus_secret_agent_call_get_secrets (priv->proxy,
 	                                      dict,
 	                                      path,
@@ -370,6 +373,7 @@ nm_secret_agent_get_secrets (NMSecretAgent *self,
 	                                      flags,
 	                                      r->cancellable,
 	                                      get_callback, r);
+	g_dbus_proxy_set_default_timeout (G_DBUS_PROXY (priv->proxy), -1);
 
 	return r;
 }
